### PR TITLE
Fix a TF Vision Encoder Decoder test

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Simple fix for the following CI failure: 

```
tests.vision_encoder_decoder.test_modeling_tf_vision_encoder_decoder.TFViT2GPT2EncoderDecoderModelTest.test_pt_tf_equivalence
```
This is due to a few PyTorch device issue (need `pt_inputs` to be on the correct device).

Tested on GCP GPU VM - it pass now.

@patrickvonplaten @sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thnaks for fixing!

**Comment by ydshieh** [comment]
Probably wait @patrickvonplaten too, but once ready, should I merge this PR myself ? (or still should be one of you to do so?)

**Comment by sgugger** [comment]
You can merge your own PRs as long as you have the approvals you want :-)

**Review by patrickvonplaten — APPROVED** [review]

File: tests/vision_encoder_decoder/test_modeling_tf_vision_encoder_decoder.py
Comment (patrickvonplaten) [inline tests/vision_encoder_decoder/test_modeling_tf_vision_encoder_decoder.py:327]: note that `detach()` is not needed here since it's wrapped in a `torch.no_grad()` but doesn't really matter so ok for me

**Pull request merged**
